### PR TITLE
feat: enable warnings for non null assertion and add test framework

### PR DESCRIPTION
**Pull request by timidak**
- enabled warning for `no-non-null-assertion`
- added tests for

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": false
+  },
+}
```
Comment (LucasFalkowsky) [inline .vscode/settings.json:5]: was this done, for coding purposes?

Comment (timidak) [reply .vscode/settings.json:5]: Yes, kinda. My eslint vscode extension fixes all errors on save, but when I want to produce an error in a file, to test it, ESLint autofixed it. So, basically I wasn't able to produce errors I wanted to test. 😸

Comment (LucasFalkowsky) [reply .vscode/settings.json:5]: Just delete the file and the PR is approved!

**Review by LucasFalkowsky — CHANGES_REQUESTED** [review]
Well done.

**Review by LucasFalkowsky — APPROVED** [review]
Even smarter!

**Pull request merged**